src/main/webapp/WEB-INF/view/notifications.jsp

```diff
@@ -25,19 +28,76 @@
     <% }} %>
   </nav>
 
+  <% List<Notification> unreadNotifications = (List<Notification>) request.getAttribute("unreadNotifications"); %>
+  <% List<Notification> readNotifications = (List<Notification>) request.getAttribute("readNotifications"); %>
+  <% int numUnreadNotifications = (int) request.getSession().getAttribute("numUnreadNotifications"); %>
+  <% int numReadNotifications = (int) request.getSession().getAttribute("numReadNotifications"); %>
+
   <div id="container">
 
     <% if(request.getAttribute("error") != null){ %>
         <h2 style="color:red"><%= request.getAttribute("error") %></h2>
     <% } %>
 
     <% if(request.getSession().getAttribute("user") != null){ %>
-    <h1>Hi <%= request.getSession().getAttribute("user") %>, you have _ notifications.</h1>
+      <% if(numUnreadNotifications==0){ %>
+        <h1>Hi <%= request.getSession().getAttribute("user") %>, you have
+        no new notifications.</h1>
+      <% } else if(numUnreadNotifications==1){%>
+        <h1>Hi <%= request.getSession().getAttribute("user") %>, you have
+        1 new notification.</h1>
+      <% } else{ %>
+        <h1>Hi <%= request.getSession().getAttribute("user") %>, you have
+        <%= numUnreadNotifications %> new notifications.</h1>
+      <% } %>
+
+      <% if(numUnreadNotifications != 0){ %>
```

Comment:
I like here how you include both the read and unread notifications to display, and allow the user to clear the notif messages themselves.